src/network_manager.rs

```diff
@@ -148,7 +148,7 @@ where
 
     pub fn send_pub<P: crate::publication::ToPayload>(
         &mut self,
-        pub_packet: &Pub<'_, P>,
+        pub_packet: Pub<'_, P>,
```

Comment:
Why this?

Reply:
This eases ownership semantics because of the new FnMut requiring mut access to the `pub_packet`

Reply:
If we didn't take ownership, we'd have to use `&mut`, which causes all kinds of angry compiler declarations about `&mut self` and `&mut pub_packet` ownership lifetimes.

Reply:
With FnOnce, we now require ownership anyways

Reply:
Correct.
I think you could in principle still do fine with a &Mut and mem::swap before calling func but it doesn't seem sensible.

But then (if we consume self) I'd try avoiding `impl topayload for [u8;N]` and instead only borrow. Otherwise I'd worry about copies (small arrays are copy).